chia/full_node/coin_store.py

```diff
@@ -311,24 +314,25 @@ async def get_coin_states_by_puzzle_hashes(
             return []
 
         coins = set()
-        puzzle_hashes_db: Tuple[Any, ...]
-        if self.db_wrapper.db_version == 2:
-            puzzle_hashes_db = tuple(puzzle_hashes)
-        else:
-            puzzle_hashes_db = tuple([ph.hex() for ph in puzzle_hashes])
-        async with self.coin_record_db.execute(
-            f"SELECT confirmed_index, spent_index, coinbase, puzzle_hash, "
-            f"coin_parent, amount, timestamp FROM coin_record INDEXED BY coin_puzzle_hash "
-            f'WHERE puzzle_hash in ({"?," * (len(puzzle_hashes) - 1)}?) '
-            f"AND (confirmed_index>=? OR spent_index>=?)"
-            f"{'' if include_spent_coins else 'AND spent_index=0'}",
-            puzzle_hashes_db + (min_height, min_height),
-        ) as cursor:
+        for puzzles in chunks(puzzle_hashes, MAX_SQLITE_PARAMETERS):
+            puzzle_hashes_db: Tuple[Any, ...]
+            if self.db_wrapper.db_version == 2:
+                puzzle_hashes_db = tuple(puzzles)
+            else:
+                puzzle_hashes_db = tuple([ph.hex() for ph in puzzles])
```

Comment:
This is a change unrelated to this PR, but maybe catch it while touching it.

```suggestion
                puzzle_hashes_db = tuple(ph.hex() for ph in puzzles)
```

Reply:
I was under the impression list-comprehension was specific to lists, and that you had to include the `[` and `]`. is the expression `f(x) for x in y` producing an iterator?
but then, the expression `[x]` is a list of `x`, even if `x` is an iterator, right?

Reply:
> I was under the impression list-comprehension was specific to lists, and that you had to include the `[` and `]`.

There are list comprehensions but also dict and set comprehensions.  Plus generator comprehensions (previously referred to as generator expressions).

> is the expression `f(x) for x in y` producing an iterator?

A small caveat.  `(f(x) for x in y)` is a generator comprehension and creates a generator which does indeed implement the iterator protocol.  If you pass it to a function as the only parameter then you can drop the parentheses for the generator comprehension and just use the existing function call parentheses like in `a_function(f(x) for x in y)`, or `tuple(f(x) for x in y)` as in the code here.

> but then, the expression `[x]` is a list of `x`, even if `x` is an iterator, right?

Yes.  To add to that, if you want to create a list and iterate it to get the elements of the list you would instead `list(the_iterable)`.